Plugin.php

```diff
@@ -52,43 +60,167 @@ public function boot()
+            });
+        }
+    }
+
+    /**
+     * Add collectors used by the Backend only
+     */
+    public function addBackendCollectors()
+    {
+        /** @var \Barryvdh\Debugbar\LaravelDebugbar $debugBar */
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+
+        if (Config::get('debugbar.collectors.backend', true)) {
+            Event::listen('backend.page.beforeDisplay', function (BackendController $controller, $action, array $params) use ($debugBar) {
+                $collector = new BackendCollector($controller, $action, $params);
+                if (!$debugBar->hasCollector($collector->getName())) {
+                    $debugBar->addCollector($collector);
+                }
+            });
+        }
+    }
+
+    /**
+     * Registers extensions in the CMS Twig environment
+     */
+    protected function registerCmsTwigExtensions()
+    {
+        $profile = new Profile;
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+
+        Event::listen('cms.page.beforeDisplay', function ($controller, $url, $page) use ($profile, $debugBar) {
             $twig = $controller->getTwig();
             if (!$twig->hasExtension(\Barryvdh\Debugbar\Twig\Extension\Debug::class)) {
                 $twig->addExtension(new \Barryvdh\Debugbar\Twig\Extension\Debug($this->app));
                 $twig->addExtension(new \Barryvdh\Debugbar\Twig\Extension\Stopwatch($this->app));
             }
+
+            if (!$twig->hasExtension(ProfilerExtension::class)) {
+                $twig->addExtension(new ProfilerExtension($profile));
+            }
         });
+
+        if (class_exists(\DebugBar\Bridge\NamespacedTwigProfileCollector::class)) {
+            $debugBar->addCollector(new \DebugBar\Bridge\NamespacedTwigProfileCollector($profile));
+        } else {
+            $debugBar->addCollector(new \DebugBar\Bridge\TwigProfileCollector($profile));
+        }
     }
 
     /**
-     * Register the
+     * Adds styling to the page
      */
-    public function register()
+    protected function registerResourceInjection()
     {
-        /*
-         * Register asset bundles
-         */
-        CombineAssets::registerCallback(function ($combiner) {
-            $combiner->registerBundle('$/winter/debugbar/assets/css/debugbar.less');
+        // Add styling
+        $addResources = function ($controller) {
+            $renderer = Debugbar::getJavascriptRenderer();
+            [$css, $js] = $renderer->getAssets();
+
+            // Switch base Debugbar CSS out with our own
+            foreach ($css as $key => $file) {
+                if (str_ends_with($file, 'vendor/barryvdh/laravel-debugbar/src/Resources/laravel-debugbar.css')) {
+                    $css[$key] = __DIR__ . '/assets/css/debugbar.css';
+                    continue;
+                }
+                if (str_contains($file, 'vendor/barryvdh/laravel-debugbar/src/Resources/laravel-debugbar-dark-mode')) {
```

Comment:
I'm thinking it might be a good idea if we start transitioning to using the Str facade directly in first party code as much as possible.

Reply:
@LukeTowers would it not be better to use the `str_*` functions as most of them have been made into core functions in PHP 8, like [`str_contains`](https://www.php.net/str_contains)?

Reply:
@bennothommo I'd prefer to use the Str facade as that retains support for older versions of PHP at the moment and they'll use the PHP functions if available.